rewrite-java/src/main/java/org/openrewrite/java/format/MinimumViableSpacingVisitor.java

```diff
@@ -174,27 +174,44 @@ public J.Return visitReturn(J.Return return_, P p) {
     public J.VariableDeclarations visitVariableDeclarations(J.VariableDeclarations multiVariable, P p) {
         J.VariableDeclarations v = super.visitVariableDeclarations(multiVariable, p);
 
-        boolean first = v.getLeadingAnnotations().isEmpty();
+        boolean hasLeadingAnnotations = !v.getLeadingAnnotations().isEmpty();
+        boolean hasModifiers = !v.getModifiers().isEmpty();
+
+        /*
+         * We need at least one space between leading annotations, otherwise we could get a run-on like "@NonNull@Generated@Serial".
+         * Note, this is applicable anywhere that annotations can exist, such as class declarations, etc.
```

Comment:
```suggestion
```

Though it is a lot more work, I think it's better to add this leadingAnnotations-fix to all LST elements with leading annotations, then having this comment here.  To show my point, I added another test to prove this problem does indeed exists for the other LST elements (like this comment says): https://github.com/openrewrite/rewrite/pull/5765/commits/bf7619c5993fa413e0187244b92cf766229d8cf4.

Reply:
As also commented in Slack: is it even really a problem though? My compiler accepts it, and the minimum vaiable spacing visitor need not look good: it's intended to wipe out formatting before the detected or configured style is applied.